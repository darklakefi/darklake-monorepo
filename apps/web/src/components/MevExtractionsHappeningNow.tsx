import MevExtractionCard from "./MevExtractionCard";
interface MevExtractionsHappeningNowProps {
  drainToday: {
    amountSol: number;
    amountUsd: number;
  };
  weekTotal: {
    amountSol: number;
    amountUsd: number;
  };
  attacksToday: {
    attacksTodayCount: number;
    attacksWeekCount: number;
  };
}

export default function MevExtractionsHappeningNow(props: MevExtractionsHappeningNowProps) {
  const { drainToday, weekTotal, attacksToday } = props;
  return (
    <div>
      <h2 className="text-3xl uppercase text-brand-30 font-primary mb-4">
        Mev Extraction <span className="text-brand-20">Happening Now</span>
      </h2>
      <div className="flex gap-4 flex-col md:flex-row">
        <MevExtractionCard>
          <h3 className="mb-4 text-lg text-brand-30">Today&apos;s Drain</h3>
          <span className="text-3xl">{drainToday.amountSol} SOL</span>
          <span className="text-lg text-brand-30">
            {drainToday.amountUsd.toLocaleString("en-US", {
              style: "currency",
              currency: "USD",
            })}{" "}
            extracted
          </span>
        </MevExtractionCard>
        <MevExtractionCard>
          <h3 className="mb-4 text-lg text-brand-30">7D Total MEV</h3>
          <span className="text-3xl">{weekTotal.amountSol} SOL</span>
          <span className="text-lg text-brand-30">
            {weekTotal.amountUsd.toLocaleString("en-US", {
              style: "currency",
              currency: "USD",
            })}{" "}
            extracted
          </span>
        </MevExtractionCard>
        <MevExtractionCard>
          <h3 className="mb-4 text-lg text-brand-30">Attacks Today</h3>
          <span className="text-3xl">{attacksToday.attacksTodayCount}</span>
          <span className="text-lg text-brand-30">{attacksToday.attacksWeekCount} this week</span>
        </MevExtractionCard>
      </div>
    </div>
  );
}
