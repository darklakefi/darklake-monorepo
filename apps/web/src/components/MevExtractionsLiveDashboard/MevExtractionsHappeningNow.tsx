import { formatMoney } from "@/utils/number";
import { useMemo } from "react";
import SummaryCard from "../SummaryCard";

interface MevExtractionsHappeningNowProps {
  drainToday: {
    amountSol: number;
    amountUsd: number;
  };
  weekTotal: {
    amountSol: number;
    amountUsd: number;
  };
  attacksToday: {
    attacksTodayCount: number;
    attacksWeekCount: number;
  };
}

export default function MevExtractionsHappeningNow(props: MevExtractionsHappeningNowProps) {
  const { drainToday, weekTotal, attacksToday } = props;

  const { drainTodayContent } = useMemo(() => {
    const solAmountFormatted = formatMoney(drainToday?.amountSol ?? 0, 5);
    const solAmountParts = solAmountFormatted.split(".");

    const formattedDrainToday = {
      solAmount: {
        wholePart: solAmountParts[0],
        fractionalPart: solAmountParts[1] ? solAmountParts[1] : "",
      },
      usdAmount: formatMoney(drainToday?.amountUsd ?? 0),
    };

    const drainTodayContent = (
      <div className="flex flex-col gap-1 py-0.5">
        <div className="text-3xl leading-8 text-brand-20">
          {formattedDrainToday.solAmount.wholePart}
          {!!formattedDrainToday.solAmount.fractionalPart && `.${formattedDrainToday.solAmount.fractionalPart} `}
          SOL
        </div>
        {formattedDrainToday.usdAmount !== "0" && (
          <div className="text-lg leading-6 text-brand-30">{formattedDrainToday.usdAmount} USDC</div>
        )}
      </div>
    );

    return { drainTodayContent };
  }, [drainToday]);

  const { weekTotalContent } = useMemo(() => {
    const solAmountFormatted = formatMoney(weekTotal?.amountSol ?? 0, 5);
    const solAmountParts = solAmountFormatted.split(".");

    const formattedWeekTotal = {
      solAmount: {
        wholePart: solAmountParts[0],
        fractionalPart: solAmountParts[1] ? solAmountParts[1] : "",
      },
      usdAmount: formatMoney(weekTotal?.amountUsd ?? 0),
    };

    const weekTotalContent = (
      <div className="flex flex-col gap-1 py-0.5">
        <div className="text-3xl leading-8 text-brand-20">
          {formattedWeekTotal.solAmount.wholePart}
          {!!formattedWeekTotal.solAmount.fractionalPart && `.${formattedWeekTotal.solAmount.fractionalPart} `}
          SOL
        </div>
        {formattedWeekTotal.usdAmount !== "0" && (
          <div className="text-lg leading-6 text-brand-30">{formattedWeekTotal.usdAmount} USDC</div>
        )}
      </div>
    );

    return { weekTotalContent };
  }, [weekTotal]);

  return (
    <div>
      <h2 className="text-3xl uppercase text-brand-30 font-primary mb-4">
        Mev Extraction <span className="text-brand-20">Happening Now</span>
      </h2>
      <div className="flex gap-4 flex-col md:flex-row">
        <div className="flex-1">
          <SummaryCard title="Today's Drain" content={drainTodayContent} />
        </div>

        <div className="flex-1">
          <SummaryCard title="7D Total MEV" content={weekTotalContent} />
        </div>

        <div className="flex-1">
          <SummaryCard
            title="Attacks Today"
            content={
              <div className="flex flex-col gap-1 py-0.5">
                <div className="text-3xl leading-8 text-brand-20">{attacksToday.attacksTodayCount}</div>
                {attacksToday.attacksWeekCount > 0 && (
                  <div className="text-lg leading-6 text-brand-30 uppercase">
                    {attacksToday.attacksWeekCount} this week
                  </div>
                )}
              </div>
            }
          />
        </div>
      </div>
    </div>
  );
}
