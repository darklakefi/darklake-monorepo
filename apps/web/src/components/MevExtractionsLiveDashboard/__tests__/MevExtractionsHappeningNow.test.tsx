import { act, render, RenderResult } from "@testing-library/react";
import MevExtractionsHappeningNow from "../MevExtractionsHappeningNow";

describe("<MevExtractionsHappeningNow />", () => {
  it("renders successfully", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(
        <MevExtractionsHappeningNow
          drainToday={{
            amountSol: 1,
            amountUsd: 11,
          }}
          weekTotal={{
            amountSol: 1,
            amountUsd: 1,
          }}
          attacksToday={{
            attacksTodayCount: 1,
            attacksWeekCount: 1,
          }}
        />,
      ));
    });

    expect(baseElement).toMatchSnapshot();
  });
});
