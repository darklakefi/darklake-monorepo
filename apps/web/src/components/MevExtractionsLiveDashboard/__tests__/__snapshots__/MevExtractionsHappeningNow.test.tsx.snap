// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MevExtractionsHappeningNow /> renders successfully 1`] = `
<body>
  <div>
    <div>
      <h2
        class="text-3xl uppercase text-brand-30 font-primary mb-4"
      >
        Mev Extraction 
        <span
          class="text-brand-20"
        >
          Happening Now
        </span>
      </h2>
      <div
        class="flex gap-4 flex-col md:flex-row"
      >
        <div
          class="flex-1"
        >
          <div
            class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
          >
            <div
              class="text-body-2 text-brand-30"
            >
              Today's Drain
            </div>
            <div
              class="flex flex-col gap-1 py-0.5"
            >
              <div
                class="text-body text-brand-20"
              >
                1
                 
                SOL
              </div>
              <div
                class="text-body-2 text-brand-30"
              >
                11
                 USDC
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
          >
            <div
              class="text-body-2 text-brand-30"
            >
              7D Total MEV
            </div>
            <div
              class="flex flex-col gap-1 py-0.5"
            >
              <div
                class="text-body text-brand-20"
              >
                1
                 
                SOL
              </div>
              <div
                class="text-body-2 text-brand-30"
              >
                1
                 USDC
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full"
          >
            <div
              class="text-body-2 text-brand-30"
            >
              Attacks Today
            </div>
            <div
              class="flex flex-col gap-1 py-0.5"
            >
              <div
                class="text-body text-brand-20"
              >
                1
              </div>
              <div
                class="text-body-2 text-brand-30"
              >
                1
                 this week
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
